### ISSUE TITLE
Task layer issue

### ISSUE THREAD

**Issue by azizaparveen** [Esri/solutions-components#989]
### Describe the bug

Seeing two issues so far

- [ ] if I select a record from the point layer, click Edit, the feature lost its selection mode in the map
- [ ] In Map layout, if I select a record and move the geometry, clickin gupdate button is doing nothing, also no console error.
![Manager_tasklayernot updating geometry](https://github.com/user-attachments/assets/2e2435be-df46-4514-ab6b-3f074a965a39)

### Expected Behavior

Selection from map should not be lost after clicking on Edit button, also Update button should update the geometry, I see the issue with attribute update as well.

### Reproduction Steps & Sample

Test map: https://arcgis4localgov2.maps.arcgis.com/apps/mapviewer/index.html?webmap=7029142de15e468cbc5953e4093af838

**Comment by azizaparveen** [comment]
cc @sarahmcdonald7

**Comment by chris-fox** [comment]
@azizaparveen, one issue I have seen in the past with the Update button not working is if my layer has contingent values configured and the value are not valid for the contingent values, then you can't update the feature and you get no error/warning message. This is because contingent value fields are not visible in the form. I was able to reproduce the same issue in the Map Viewer.

I notice the Trees layer in your web map has contingent values enabled for the Tree species information. I am wondering if this might be the cause.

**Comment by azizaparveen** [comment]
@chris-fox yeah, as you said I can repro this in map viewer as well. In my form, I don't have any field that has contingent values though, and it still do not update. I would expect it to work if the form do not have any field with contingent values.,
in the screenshot below there are field groups that has contingent values but I haven't used any of them in the form
![image](https://github.com/user-attachments/assets/26a7da81-c16c-46da-8e1b-fc96618d2d6b))

**Comment by chris-fox** [comment]
Yes, i found the same. In my case even when I didn't have them in the form the edit widget would not allow me to edit the record. I found if I added them to my form, it would give me an error related to the contingent values and so at least i could see and resolve the issue and then update worked. You could try adding them to the form and see if it changes the behavior, are you seeing an error related to the contingent values that is preventing the update?

**Comment by azizaparveen** [comment]
Submitted an issue in JS API repo https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/66306 for fixing editing issue

**Comment by azizaparveen** [comment]
Did little more testing on this today. I published a layer with contingent value fields without enabling task and see the same issue sometimes (but cannot consistently repro always), so as Chris said it may not task layer specific.

Right now we don't have much to do except waiting for JS API (https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/66306) to fix it. In case, they fixed it in R3, here are test apps .

Devext: https://solutions.mapsdevext.arcgis.com/apps/instant/manager/index.html?appid=7ff595ffea994941a119895b1bb7df58
Qa: https://solutionsqa.mapsqa.arcgis.com/apps/instant/manager/index.html?appid=c9cda33b25a84dcebc558b83c36537f8

**Comment by azizaparveen** [comment]
This is being tracked in this https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/56350 issue in JS API repo, looks like the milestone is set for 4.32, next release

**Comment by azizaparveen** [comment]
This is moved to R2, 4.33

**Comment by azizaparveen** [comment]
This issue is now in backlog of JS API repo, no estimated timeline for the fix.
